src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -146,13 +146,14 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
           clusterMonitorSupervisor ! ClusterCreated(savedCluster)
           savedCluster
         }
-        clusterFuture.onComplete {
-          case Failure(_) =>
-            //make a dummy cluster with the details
-            val clusterToDelete = Cluster.createDummyForDeletion(clusterRequest, userEmail, clusterName, googleProject, serviceAccountInfo)
-            internalDeleteCluster(userEmail, clusterToDelete) //don't wait for it
```

Comment:
This is sort of the key: if a create cluster request fails, we should clean up resources in Google/Sam but _not_ kick off the cluster deletion monitor. The monitor is more for keeping the Leo DB state in sync with Google, but in this case the failed cluster is not persisted in the DB. Previously it was possible for a user to create a cluster, get an error, retry, and end up with 2 monitor actors for the same cluster which caused the "stuck in Creating" issue.